tests/e2e/e2e_suite_test.go

```diff
@@ -58,224 +53,285 @@ const (
-		k8SClient:   framework.NewKubeClient(k8sClientset),
-		k8sConfig:   k8sConfig,
-		helmClient:  helmClient,
-		suiteConfig: suiteCfg,
-	}
-}
+		BeforeAll(func(ctx context.Context) {
+			if testContext.kubeconfig == "" {
+				_, _ = fmt.Fprintln(GinkgoWriter, "kubeconfig parameter is empty. Defaulting to ~/.kube/config")
+			}
 
-func (s *Suite) SetupSuite() {
-	defer func() {
-		// This is only run if SetupSuite fails or panics, otherwise TearDownSuite will call it
-		if !s.setupDone {
-			s.TearDownSuite()
-		}
-	}()
-	s.T().Log("Starting E2E test setup...")
-
-	s.labels = map[string]string{
-		"e2eRunID": runID.String(),
-	}
-
-	s.T().Logf("Creating namespace: %s...", s.namespace)
-
-	_, err := s.k8SClient.CreateNamespace(s.ctx, s.namespace, s.labels)
-	s.Require().NoError(err, "Failed to create namespace")
-
-	s.T().Logf("Namespace: %q created", s.namespace)
-
-	s.T().Logf("Installing the helm chart: %s", s.chart)
-
-	values := []string{
-		fmt.Sprintf("serviceMonitor.enabled=%v", false),
-	}
-
-	if s.arguments != "" {
-		values = append(values, fmt.Sprintf("arguments=%s", s.arguments))
-	}
-
-	if s.imageRepository != "" {
-		values = append(values, fmt.Sprintf("image.repository=%s", s.imageRepository))
-	}
-	if s.imageTag != "" {
-		values = append(values, fmt.Sprintf("image.tag=%s", s.imageTag))
-	}
-
-	s.helmReleaseName, err = s.helmClient.Install(s.ctx, values, framework.HelmChartOptions{
-		CleanupOnFail: true,
-		GenerateName:  true,
-		Timeout:       5 * time.Minute,
-		Wait:          true,
-		DryRun:        false,
-	})
+			if len(testContext.chart) == 0 {
+				Fail("chart parameter is empty")
+			}
 
-	s.Require().NoError(err, "Cannot install helm chart: %s", s.chart)
+			var err error
 
-	s.T().Logf("Helm chart: %q installed", s.chart)
+			testContext.kubeconfig, err = framework.ResolvePath(testContext.kubeconfig)
+			Expect(err).ShouldNot(HaveOccurred(),
+				"cannot resolve path to kubeconfig: %s, err: %v", testContext.kubeconfig, err)
 
-	s.T().Log("Test setup ready.")
+			if _, err := os.Stat(testContext.kubeconfig); os.IsNotExist(err) {
+				Fail(fmt.Sprintf("kubeconfig file does not exist: %s", testContext.kubeconfig))
+			}
 
-	s.setupDone = true
-}
+			// Init out-of-cluster K8S client
+			k8sConfig, err := clientcmd.BuildConfigFromFlags("", testContext.kubeconfig)
+			Expect(err).ShouldNot(HaveOccurred(), "unable to load kubeconfig from %s; err: %s", testContext.kubeconfig, err)
+
+			k8sClientset, err := kubernetes.NewForConfig(k8sConfig)
+			Expect(err).ShouldNot(HaveOccurred(), "cannot create k8s client: %s", err)
+
+			kubeClient = framework.NewKubeClient(k8sClientset)
+
+			labels = map[string]string{
+				"e2eRunID": runID.String(),
+			}
+
+			_, _ = fmt.Fprintf(GinkgoWriter, "Creating namespace: %q started.\n", testContext.namespace)
+
+			_, err = kubeClient.CreateNamespace(ctx, testContext.namespace, labels)
+			Expect(err).ShouldNot(HaveOccurred(), "Creating namespace: failed")
```

Comment:
@rohit-arora-dev , I thought about this. I decided that it would be less harmful if we failed in case the namespace existed. It is a safeguard from accidentally running the e2e test on an existing namespace.